class/practice2.2/main.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <sstream>
+#include <algorithm>
+
+using namespace std;
+
+struct Pixel {
+    int r, g, b;
+};
+
+bool isValidPixel(const Pixel &pixel) {
+    return (pixel.r >= 0 && pixel.r <= 255) && (pixel.g >= 0 && pixel.g <= 255) && (pixel.b >= 0 && pixel.b <= 255);
+}
+
+bool loadPixels(const string &filename, vector<vector<Pixel>> &pixels, string &errorMessage) {
```

Comment:
As you can see we have output parameter in this function pixels, and the same structure is used for savePixels and other functions.
If we will review that functions all of them perform some action on that data. Therefore it is reasonable to encapsulate that data into the class with corresponding methods.
```
class Image {
public:
     bool initFromFile(const std::string& filename) {

    }

    bool saveToFile(const std::string& outFile) {
   }
...


private:
     std::vector<std::vector<Pixel>> pixels;
}